examples/formatting/custom/src/formatter.js

```diff
@@ -3,16 +3,64 @@ import MessageFormat from 'messageformat'
 export default class CustomFormatter {
   constructor (options = {}) {
     this._locale = options.locale || 'en-US'
-    this._formatter = new MessageFormat(this._locale)
+    this._formatter = new MessageFormat(this._locale, {returnType: 'values'})
     this._caches = Object.create(null)
   }
 
-  interpolate (message, values) {
+  interpolate (message, values, path) {
     let fn = this._caches[message]
     if (!fn) {
       fn = this._formatter.compile(message, this._locale)
       this._caches[message] = fn
     }
-    return [fn(values)]
+
+    if (values !== null && values !== undefined) {
+      /**
+       * Component interpolation in the following format passes `values` as an Array-like `Object`
```

Comment:
Is this an expected behaviour/known issue of `vue-i18n` or should I report this as a bug?